Make error when hash doesn't match more informative

```
    def _check_download_hash(self, fname, downloaded):
        """
        Check the hash of the downloaded file against the one in the registry.
    
        Parameters
        ----------
        fname : str
            The file name in the registry.
        downloaded : str
            The pull path to the downloaded file.
    
        Raises
        ------
        :class:`ValueError`
            If the hashes don't match.
    
        """
        tmphash = file_hash(downloaded)
        if tmphash != self.registry[fname]:
            raise ValueError(
                "Hash of downloaded file '{}' doesn't match the entry in the registry:"
                " Expected '{}' and got '{}'.".format(
>                   downloaded, self.registry[fname], tmphash
                )
            )
E           ValueError: Hash of downloaded file '/home/mark/.cache/scikit-image/tmpydrsiovh' doesn't match the entry in the registry: Expected '8cd81c5fccdbcca6b623a5f157e71b27e91907e667626a0e07da279745e12d19' and got '03097789a3dcbb0e40d20b9ef82537dbc3b670b6a7f2268d735470f22e003a91'.
```

I think the `fname` should be printed, at least that is probably more related than a tempfile name

--- a/pooch/core.py
+++ b/pooch/core.py
@@ -455,9 +455,9 @@ class Pooch:
         tmphash = file_hash(downloaded)
         if tmphash != self.registry[fname]:
             raise ValueError(
-                "Hash of downloaded file '{}' doesn't match the entry in the registry:"
+                "Hash of downloaded file '{}' doesn't match the entry in the registry."
                 " Expected '{}' and got '{}'.".format(
-                    downloaded, self.registry[fname], tmphash
+                    fname, self.registry[fname], tmphash
                 )
             )
 

